### PR TITLE
Add documentation for alter_data_node

### PR DESCRIPTION
**Pull request by nikkhils**
# Description

Add documentation for alter_data_node 

# Links

# Writing help

For information about style and word usage, see the [style guide](https://docs.timescale.com/timescaledb/latest/contribute-to-docs/)

# Review checklists
Reviewers: use this section to ensure you have checked everything before approving this PR:

## Subject matter expert (SME) review checklist

- [ ] Is the content technically accurate?
- [ ] Is the content complete?
- [ ] Is the content presented in a logical order?
- [ ] Does the content use appropriate names for features and products?
- [ ] Does the content provide relevant links to further information?

## Documentation team review checklist

- [ ] Is the content free from typos?
- [ ] Does the content use plain English?
- [ ] Does the content contain clear sections for concepts, tasks, and references?
- [ ] Have any images been uploaded to the correct location, and are resolvable?
- [ ] If the page index was updated, are redirects required
      and have they been implemented?
- [ ] Have you checked the built version of this content?

### CODE REVIEW

File: api/alter_data_node.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+### Sample usage
+
+To change the port number and host information for an existing data node `dn1`:
+
+```sql
+SELECT alter_data_node('dn1', host => 'dn1.example.com', port => 6999);
+```
+
+By default a data node is available for read/write queries. Reads/writes start erroring
+out if the data node becomes unavailable for whatever reasons. This api provides an
```
Comment (Loquacity) [inline api/alter_data_node.md:68]:
```suggestion
Data nodes are available for read/write queries by default. If the data 
node becomes unavailable for some reason, the read/write query gives 
an error. This API provides an
```

File: api/alter_data_node.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+optional argument `available` to mark an existing data node as available/unavailable
```
Comment (Loquacity) [inline api/alter_data_node.md:69]:
```suggestion
optional argument, `available`, to mark an existing data node as available or unavailable
```

File: timescaledb/how-to-guides/multinode-timescaledb/multinode-ha.md [hunk 3]
```diff
@@ -103,14 +103,44 @@ includes one replica of each chunk in the query plan.
...
-while a data node is not available, its chunks would diverge from
-other replicas in the cluster. If waiting for the data node to come
-back is not an option, either because it takes to long or the node is
-permanently failed, one can delete it instead. To be able to delete a
-data node, all of its chunks must have at least one replica on other
-data nodes. For example:
+node result in an error. This is to preserve data consistency in
+case the data node becomes available again. You can use the
+[`alter_data_node`][alter_data_node] call to mark a failed data node
+as unavailable by running the following:
```
Comment (Loquacity) [inline timescaledb/how-to-guides/multinode-timescaledb/multinode-ha.md:109]:
```suggestion
as unavailable by running this query:
```

File: timescaledb/how-to-guides/multinode-timescaledb/multinode-ha.md [hunk 4]
```diff
@@ -103,14 +103,44 @@ includes one replica of each chunk in the query plan.
...
+
+```sql
+SELECT alter_data_node('data_node_2', available => false);
+```
+
+Setting `available => false` means that the data node is no longer
+used for reads and writes queries.
+
+To fail over reads, the [`alter_data_node`][alter_data_node]  call finds
```
Comment (Loquacity) [inline timescaledb/how-to-guides/multinode-timescaledb/multinode-ha.md:118]:
```suggestion
To fail over reads, the [`alter_data_node`][alter_data_node] call finds
```

File: timescaledb/how-to-guides/multinode-timescaledb/multinode-ha.md [hunk 5]
```diff
@@ -103,14 +103,44 @@ includes one replica of each chunk in the query plan.
...
+all the chunks for which the unavailable data node is the **primary** query
```
Comment (Loquacity) [inline timescaledb/how-to-guides/multinode-timescaledb/multinode-ha.md:119]:
```suggestion
all the chunks for which the unavailable data node is the primary query
```

File: timescaledb/how-to-guides/multinode-timescaledb/multinode-ha.md [hunk 6]
```diff
@@ -103,14 +103,44 @@ includes one replica of each chunk in the query plan.
...
+target and fails over to a chunk replica on another data node instead.
```
Comment (Loquacity) [inline timescaledb/how-to-guides/multinode-timescaledb/multinode-ha.md:120]:
```suggestion
target and fails over to a chunk replica on another data node.
```

File: timescaledb/how-to-guides/multinode-timescaledb/multinode-ha.md [hunk 7]
```diff
@@ -103,14 +103,44 @@ includes one replica of each chunk in the query plan.
...
+However, if some chunks do not have a replica to fail over to, a warning
+is raised. Reads continue to fail for chunks that do not have a chunk
+replica on any other data nodes.
+
+To fail over writes, any activity that intends to write to the failed
+node marks the involved chunk as **stale** for this specific failed
```
Comment (Loquacity) [inline timescaledb/how-to-guides/multinode-timescaledb/multinode-ha.md:126]:
```suggestion
node marks the involved chunk as stale for the specific failed
```

File: timescaledb/how-to-guides/multinode-timescaledb/multinode-ha.md [hunk 8]
```diff
@@ -103,14 +103,44 @@ includes one replica of each chunk in the query plan.
...
+node by changing the metadata on the access node. This is only done
+for natively replicated chunks. This allows you to continue to write
+to other chunk replicas on other data nodes while this failed node has
```
Comment (Loquacity) [inline timescaledb/how-to-guides/multinode-timescaledb/multinode-ha.md:129]:
```suggestion
to other chunk replicas on other data nodes while the failed node has
```

File: timescaledb/how-to-guides/multinode-timescaledb/multinode-ha.md [hunk 9]
```diff
@@ -103,14 +103,44 @@ includes one replica of each chunk in the query plan.
...
+been marked as unavailable. Writes continue to fail for chunks that do
+not have a chunk replica on any other data nodes. Also note that chunks
+on the failed node which do not get written into are not affected in any
+way.
```
Comment (Loquacity) [inline timescaledb/how-to-guides/multinode-timescaledb/multinode-ha.md:133]:
```suggestion
on the failed node which do not get written into are not affected.
```

File: timescaledb/how-to-guides/multinode-timescaledb/multinode-ha.md [hunk 10]
```diff
@@ -103,14 +103,44 @@ includes one replica of each chunk in the query plan.
...
+
+The act of marking a chunk as **stale** means that the chunk becomes
```
Comment (Loquacity) [inline timescaledb/how-to-guides/multinode-timescaledb/multinode-ha.md:135]:
```suggestion
When you mark a chunk as stale, the chunk becomes
```

File: timescaledb/how-to-guides/multinode-timescaledb/multinode-ha.md [hunk 11]
```diff
@@ -103,14 +103,44 @@ includes one replica of each chunk in the query plan.
...
+under-replicated. When the failed data node becomes available then such
+chunks can be re-balanced by using the [`copy_chunk`][copy_chunk] api as
+explained below.
```
Comment (Loquacity) [inline timescaledb/how-to-guides/multinode-timescaledb/multinode-ha.md:138]:
```suggestion
chunks can be re-balanced using the [`copy_chunk`][copy_chunk] API.
```

File: timescaledb/how-to-guides/multinode-timescaledb/multinode-ha.md [hunk 12]
```diff
@@ -128,13 +158,13 @@ You cannot force the deletion of a data node if it would mean that a multi-node
 cluster permanently loses data.
 </highlight>
 
-When you have successfully removed a failed data node, some data
-chunks might lack replicas, but queries and inserts work as normal
-again. However, the cluster stays in a vulnerable state until all
-chunks are fully replicated again.
+When you have successfully removed a failed data node or marked a
```
Comment (Loquacity) [inline timescaledb/how-to-guides/multinode-timescaledb/multinode-ha.md:161]:
```suggestion
When you have successfully removed a failed data node, or marked a
```

File: timescaledb/how-to-guides/multinode-timescaledb/multinode-ha.md [hunk 13]
```diff
@@ -128,13 +158,13 @@ You cannot force the deletion of a data node if it would mean that a multi-node
...
+failed data node unavailable, some data chunks might lack replicas, but
```
Comment (Loquacity) [inline timescaledb/how-to-guides/multinode-timescaledb/multinode-ha.md:162]:
```suggestion
failed data node unavailable, some data chunks might lack replicas but
```

File: timescaledb/how-to-guides/multinode-timescaledb/multinode-ha.md [hunk 14]
```diff
@@ -128,13 +158,13 @@ You cannot force the deletion of a data node if it would mean that a multi-node
...
+queries and inserts work as normal again. However, the cluster stays in
+a vulnerable state until all chunks are fully replicated again.
```
Comment (Loquacity) [inline timescaledb/how-to-guides/multinode-timescaledb/multinode-ha.md:164]:
```suggestion
a vulnerable state until all chunks are fully replicated.
```

File: timescaledb/how-to-guides/multinode-timescaledb/multinode-ha.md [hunk 15]
```diff
@@ -128,13 +158,13 @@ You cannot force the deletion of a data node if it would mean that a multi-node
...
 
-When you have restored a failed data node, you can see the chunks that need to
-be replicated with this query:
+When you have restored a failed data node or marked it available again, you can
+see the chunks that need to be replicated with this query:
 
 <!--- Still experimental? --LKB 2021-10-20-->
```
Comment (Loquacity) [inline timescaledb/how-to-guides/multinode-timescaledb/multinode-ha.md:169]: I think this question from a year ago still stands ... 🤔

File: api/alter_data_node.md [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+```sql
+SELECT alter_data_node('dn1', host => 'dn1.example.com', port => 6999);
+```
+
+Data nodes are available for read/write queries by default. If the data node
+becomes unavailable for some reason, the read/write query gives an error. This
+API provides an optional argument, `available`, to mark an existing data node
+as available or unavailable for read/write queries. By marking a data node as
+unavailable you can allow read/write queries to proceed in the cluster as
+described in the [multi-node HA section][multi-node-ha]
```
Comment (Loquacity) [inline api/alter_data_node.md:71]:
```suggestion
unavailable you can allow read/write queries to proceed in the cluster. 
For more information, see the [multi-node HA section][multi-node-ha].
```

**Review by Loquacity — APPROVED** [review]
Just one non-blocking suggestion.

File: api/alter_data_node.md (hunk 16 above)
Comment (nikkhils) [reply api/alter_data_node.md]: Done.

**Pull request merged**